### PR TITLE
DAOS-11260 dfuse: Apply timeouts to dfuse data cache.

### PR DESCRIPTION
**Pull request by ashleypittman**
Use the information we keep on when files are opened to track
how old any data in the kernel page cache might be and only
set the keep_cache flag if it's recent.  Extend the current
container setting from being an on/off to also accepting a value.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/client/dfuse/dfuse_core.c [hunk 1]
```diff
@@ -674,7 +678,7 @@ dfuse_set_default_cont_cache_values(struct dfuse_cont *dfc)
 	dfc->dfc_dentry_timeout     = 1;
 	dfc->dfc_dentry_dir_timeout = 5;
 	dfc->dfc_ndentry_timeout    = 1;
-	dfc->dfc_data_caching       = true;
+	dfc->dfc_data_timeout       = 60 * 10;
```
Comment (mchaarawi) [inline src/client/dfuse/dfuse_core.c:681]: curious, why 600 here?

**Review by mchaarawi — APPROVED** [review]
looks fine to me, just had one question in the comments.
i think it would be good for me to try this with my ai benchmark just to see if it still behaves as expected in regards to data caching. won't be able to get to this though till next week.

File: src/client/dfuse/dfuse.h [hunk 2]
```diff
@@ -563,6 +569,9 @@ struct dfuse_inode_entry {
 	d_list_t                 ie_htl;
 
 	/* Time of last kernel cache update.
+	 * For directories this is the time of the most recent closedir for a handle which may
+	 * have populated the cache.
+	 * For files this is when the file was last closed after been written to.
```
Comment (daltonbohning) [inline src/client/dfuse/dfuse.h:574]:
```suggestion
	 * For files this is when the file was last closed after being written to.
```

**Review by daltonbohning — APPROVED** [review]

File: src/client/dfuse/dfuse_core.c (hunk 1 above)
Comment (ashleypittman) [reply src/client/dfuse/dfuse_core.c:681]: We've got a draw a line somewhere, I wanted something that would help and be meaningful for many use-cases but that wasn't forever.

File: src/client/dfuse/dfuse.h (hunk 2 above)
Comment (ashleypittman) [reply src/client/dfuse/dfuse.h:574]: I've pulled these two grammar corrections into another PR, will update in due course.

**Pull request merged**